pkg/tsdb/jaeger/client.go

```diff
@@ -92,6 +95,82 @@ func (j *JaegerClient) Operations(s string) ([]string, error) {
 	return operations, err
 }
 
+func (j *JaegerClient) Search(query *JaegerQuery) ([]TraceResponse, error) {
+	jaegerURL, err := url.Parse(j.url)
+	if err != nil {
+		return []TraceResponse{}, fmt.Errorf("failed to parse Jaeger URL: %w", err)
+	}
+	jaegerURL.Path = "/api/traces"
+
+	if query.Tags != "" {
+		tagMap := make(map[string]string)
+		pairs := strings.Split(query.Tags, " ")
+		for _, pair := range pairs {
+			kv := strings.SplitN(pair, "=", 2)
+			if len(kv) != 2 {
+				continue
+			}
+			tagMap[kv[0]] = kv[1]
```

Comment:
We should probably use here logfmt package to make this more resiliant to extra spaces and quotes and so on. Something like:
```
	if query.Tags != "" {
		tagMap := make(map[string]string)
		decoder := logfmt.NewDecoder(strings.NewReader(query.Tags))
		for decoder.ScanRecord() {
			for decoder.ScanKeyval() {
				key := decoder.Key()
				value := decoder.Value()
				tagMap[string(key)] = string(value)
			}
		}

		marshaledTags, err := json.Marshal(tagMap)
....
```

Reply:
Also would be worth adding couple of tests for parsing this tags.